### PR TITLE
Feature: add in style parameter

### PR DESCRIPTION
**Pull request by TRex22**
# What does this PR do?
- Implements the ability to modify the console styles of the progress bar
- List the available styles
- Specs
- Readme

# Available Styles
image

### CODE REVIEW

File: README.mkd [hunk 1]
```diff
@@ -51,6 +51,18 @@ pass a number to `#increment!`
 
     bar.increment! 42
 
+## Setting the Style
```
Comment (TRex22) [inline README.mkd]: I ended up going with `style` ... could also use `formatting` / `format` 🤔

File: lib/progress_bar/styles.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
+class ProgressBar
```
Comment (TRex22) [inline lib/progress_bar/styles.rb]: So Im aware and have used gems which do this for you. But I didnt want to introduce a new dependency and this allows for more internal control by the library as to working styles.

I tested this on Ubuntu 18.04 and Mac OSX (Iterm2).

**Comment by paul** [comment]
This change I'm less excited about. It seems like a fairly complicated feature and I'd much prefer keep this gem as light as possible. I might be open to having a separate "theme" accessor, that's either a hash or a ProgressBar::Theme struct, which contains colors and delimiters (from your previous PR).

It might work something like this:

```ruby
MY_BAR_THEME = {
  foreground: :green # or 32
  background: :black # or a number
  bar_character: "#",
  delimiter: "[]"
}

bar = ProgressBar.new(100, theme: MY_BAR_THEME)

big_list.each do |item|
  do_stuff
rescue 
  bar.theme[:foreground] = :red
  # or bar.theme = MY_ERROR_THEME
ensure
  bar.increment!
end
```

Alternatively, instead of ProgressBar having to know anything about color, there could just be `:prefix` and `:suffix` attributes on the theme, and you could set them to `\e[32m` or whatever other terminal colors you want. The same could be done with the delimiters, like `delimiters: ["\e[36m[\e[38;5;240m", "\e36m]\e0m"]`, which would give even more control over how to make the bar look.

image

WDYT?

**Comment by TRex22** [comment]
I initially didnt like the second idea but its growing on me. It would allow anyone using this gem to decide and use their own modifiers. It would also mean a hardcoded set of styles based on the limited default console set wouldnt be needed. 

I wouldnt go for the first idea because I feel its even more restrictive than the first. I would rather leave style decisions to whomever is using the library.

I will update this to use the `prefix` `suffix` idea but have a bit more of an extensive readme section to help anyone get started. Such as including the screenshot above.

**Comment by TRex22** [comment]
Im going to close this PR and re-do it using the `prefix` and `suffix` ideas

**Pull request closed without merging**